### PR TITLE
Add TTP Parser with template overloads from Git

### PR DESCRIPTION
**Pull request by jeffkala**
- Added TTP Parser support.
- Added ability to load TTP templates from Git Repository.
- Added TTP template precedence loading from Git Repository.
- Added Sync Device from Network support for Palo Alto Panos.
- Added a ability to use nautobot-app-nornir connection option extras in Sync Devices job.

### CODE REVIEW

**Comment by jeffkala** [comment]
completes #275

**Comment by jeffkala** [comment]
@scottpeterman FYI as a follow up from Autocon2

**Comment by jeffkala** [comment]
@jeffkala still to add Docs around the new functionality to inject nornir settings (e.g fast_cli) into the processor.

**Review by susanhooks — APPROVED** [review]
I'm good with this.

File: docs/dev/extending.md [hunk 1]
```diff
@@ -20,3 +20,27 @@ Extending the platform support for the SSoT specific jobs should be accomplished
 New platform support should be simplified in this framework, by providing a YAML file.
 
 The format of these YAML files are and how to extend this application is covered in [App YAML Overrides](../user/app_yaml_overrides.md).
+
+### Passing Custom Nornir Connection Options
+
+Device Onboarding 4.0 uses Netmiko as the automation engine that queries the devices for information; more specifically, nornir-netmiko. To extend the device onboarding app to pass `extras` to the connection options the following can be added to `nautobot_plugin_nornir` `PLUGIN_CONFIG`.
```
[the remark this answers is not in the document]
Comment (jeffkala) [reply docs/dev/extending.md:26]: will leave that to you, happy to move it.

[the remark this answers is not in the document]
Comment (scetron) [reply docs/dev/extending.md:26]: Yes could you move this to the User Guide > Using the App. Thank you.

[the remark this answers is not in the document]
Comment (jeffkala) [reply docs/dev/extending.md]: pushed up now.

**Pull request merged**